code/modules/mob/living/basic/bots/cleanbot/cleanbot_ai.dm

```diff
@@ -77,11 +76,13 @@
 	for(var/atom/found_item in found)
 		if(QDELETED(controller.pawn))
 			break
-		if(LAZYACCESS(ignore_list, REF(found_item)))
+		if(LAZYACCESS(ignore_list, found_item))
 			continue
+		if(get_turf(found_item) == get_turf(controller.pawn))
+			return found_item
 		var/list/path = get_path_to(controller.pawn, found_item, max_distance = BOT_CLEAN_PATH_LIMIT, access = controller.get_access())
 		if(!length(path))
-			controller.set_blackboard_key_assoc_lazylist(BB_TEMPORARY_IGNORE_LIST, REF(found_item), TRUE)
+			controller.set_blackboard_key_assoc_lazylist(BB_TEMPORARY_IGNORE_LIST, found_item, TRUE)
```

Comment:
Why was this changed from `ref`s anyways? I thought you have a use for it but I don't see one

Reply:
i initially used refs because i was worried about hard deletes, but upon studying the code i discovered that set_blackboard_key_assoc_lazylist handles the cleanup through signals so i figured it was unnecessary to use refs which are a bit more expensive to make

Reply:
Fair enough